Library/macOSRichTextEditor/Source/RichTextEditor.m

```diff
@@ -1052,6 +1060,57 @@ - (NSFont *)fontwithBoldTrait:(NSNumber *)isBold italicTrait:(NSNumber *)isItali
 	return newFont;
 }
 
+/**
+ * Does not allow cursor to be right beside the bullet point.
+ *
+ * @param beginRange The beginning position of the paragraph
+ * @param currentRange The current cursor position
+ */
+- (NSRange)moveCursorAroundBulletListIfApplicableWithBeginningLocation:(NSRange)beginRange andPrevious:(NSRange)previousRange andCurrent:(NSRange)currentRange isMouseClick:(BOOL)mouse {
+    NSUInteger previous = self.previousCursorPosition;
+    NSUInteger begin = beginRange.location;
+    NSUInteger current = currentRange.location;
+    NSRange finalRange = currentRange;
+    if (self.justDeletedBackward) {
+        return finalRange;
+    }
+    
+    BOOL currentParagraphHasBullet = [[self.attributedString.string substringFromIndex:begin] hasPrefix:self.BULLET_STRING];
+    if (currentParagraphHasBullet) {
+        if (!mouse && (current == begin + 1)) { // select bullet point when using keyboard arrow keys
+            if (previousRange.location > current) {
+                finalRange = NSMakeRange(begin, currentRange.length + 1);
+            }
+            else if (previousRange.location < current) {
+                finalRange = NSMakeRange(current + 1, currentRange.length - 1);
+            }
+        }
+        else if (begin == 0 && current <= 1) { // move cursor back to position 2
+            finalRange = currentRange.length >= 1 ? NSMakeRange(begin, finalRange.length + 1) : NSMakeRange(2, 0);
+        }
+        else if (begin > 0) {
+            if ((current == begin && (previous > current || previous < current)) ||
+                (current == (begin + 1) && (previous < current || current == previous))) { // cursor moved from in bullet to front of bullet
+                finalRange = currentRange.length >= 1 ? NSMakeRange(begin, finalRange.length + 1) : NSMakeRange(begin + 2, 0);
+            }
+            else if (current == (begin + 1) && previous > current) { // cursor moved from in bullet to beside of bullet
+                finalRange = currentRange.length >= 1 ? NSMakeRange(begin, finalRange.length + 1) : NSMakeRange(begin - 1, 0);
+            }
+            else if ((current == begin) && (begin == previous)) {
+                finalRange = currentRange.length >= 1 ? NSMakeRange(begin, finalRange.length + 1) : NSMakeRange(begin + 2, 0);
+            }
+        }
+    }
+    else {
+//        if (!mouse && [[self.attributedString.string substringFromIndex:begin] rangeOfString:@"\u2022"].location != NSNotFound) {
+//            NSLog(@"%@", [self.attributedString.string substringFromIndex:begin]);
+//        }
+        // TODO: select bullet point totally from outside the bullet point
```

Comment:
What is the purpose of this empty commented `else` statement?

Reply:
As of right now, it has no function. Its for being able to select the whole bullet from another bullet list or just another paragraph

Reply:
Then, for the purposes of good open source practice, this block of code should not belong in this PR, especially when it is marked with a `TODO:`, since that heavily implies the PR is not ready. 

As an FYI, for PRs that are looking for feedback but aren't done, common practice is to mark them as WIP: [title] in the title.